collector/src/main/java/io/prometheus/jmx/util/map/MapAccessor.java

```diff
@@ -0,0 +1,244 @@
+/*
+ * Copyright (C) 2022-2023 The Prometheus jmx_exporter Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.prometheus.jmx.util.map;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Supplier;
+import java.util.regex.Pattern;
+
+/**
+ * Class to implement a MapAccessor to work with nested Maps / values
+ */
+@SuppressWarnings("unchecked")
+public class MapAccessor {
```

Comment:
* `MapAccessor` does not need to be in the `collector` module, it should be moved to `jmx_prometheus_common`.
* `MapAccessor` is a very generic name, and it's hard to see why this is there. What about renaming to `YamlMapAccessor`?
* Packages `io.prometheus.jmx.common.util.function` and `io.prometheus.jmx.util.map` are all about parsing the YAML config. What about moving all classes to a `io.prometheus.jmx.common.yaml` or `io.prometheus.jmx.common.config` package?

Reply:
Renamed `MapAccessor` to `YamlMapAccessor` and moved it `io.prometheus.jmx.common.yaml` in the `jmx_prometheus_common` project.